### ISSUE TITLE
Simulation is too slow

### ISSUE THREAD

**Issue by githubpsyche** [githubpsyche/jaxcmr#18]
I'm currently called jax functions trial by trial. For the PEERS dataset, this results in a runtime of about 8 minutes for Base CMR.

 Multitrial simulation that can handle item_count diversity is a key next step.

**Comment by githubpsyche** [comment]
```python
simulate_trials = jax.vmap(
        jax.vmap(simulate_trial, in_axes=(None, None, 0, None)),
        in_axes=(None, None, None, 0)
    )
```

This chain of vmaps produces a param_count x rng_count x recall_count array, but it has issues. 

It probably uses the same rng for multiple simulations -- once per parameter set. 
Yup, added a test confirming as much.

Okay, so we need a different solution. Probably add an argument to a dispatch to simulate_trial that takes an experiment_count. Then split(rng, experiment_count) internally to enforce independence.

**Comment by githubpsyche** [comment]
Even then, any vmap over a function that takes rng as arguments needs to also map over a split rng in order to maintain independence between trials. Oh well!

**Comment by githubpsyche** [comment]
We'll next want to confirm the appropriate simulation of trials with variable presentations...

**Comment by githubpsyche** [comment]
Okay, great. vmapping will add another dimension to the result ( param_count x experiment_count x trial_count x recall_count array), but I'm okay with that.

Now how do I fit this into full-on dataset simulation? What are the missing pieces?

Relevant arguments in my simulate fn:

```
result = simulate_h5_from_h5(
    BaseCMR.create, #Instance_CMR, #Base_CMR, # a free recall model
    param_list, #parameters.fixed, # parameters for model initialization
    data, # data in EMBAM format 
    generate_trial_mask(data, trial_query), # mask to filter data with before simulation 
    experiment_count=100, # number of simulations to run
    ignore_first_recall=ignore_first_recall, # whether to use first recalls from data instead of simulating them
    embedding_list=None, # strings embeddings to use for semantic associations
)
```

Let's review what each argument is for:

1. **BaseCMR.create.** This is selecting the model to simulate data with.
2. **param_list**. This is the fitted parameters for each subject that I will configure the model with. Also identifies each subject.
3. **data**. Has a lot of metadata I want to follow. For every trial, simulated dataset should have matching annotations. I use repmat to help match simulated trials to these annotations.
4. **trial mask**. Filter to apply across data matrices to select trials for simulation.
5. **experiment_count**. Number of times to simulate the dataset.
6. **ignore_first_recall**. 
7. **embedding_list**. When data includes an string index for each item, this embedding list provides the matching embedding for that item at that index.

So along with simulating data for each trial, and managing heterogenous item counts, I also need to store this results in a matrix dict in a way that tracks data attributes. 

With a >1 experiment count, this normally involves a call to repmat. 

I'll further need a reshape given my simulation method.

And before that, I need to filter trials based on the provided mask.

We'll ignore 6 and 7 for now.

**Comment by githubpsyche** [comment]
We can ignore 5 too, at first.

**Comment by githubpsyche** [comment]
Great. So now I can simulate all needed data in about a minute. And the code is cleaner. Was this worth it? It could prove to be.
